db/migrate/20210930182050_modify_place_admin_levels.rb

```diff
@@ -0,0 +1,45 @@
+class ModifyPlaceAdminLevels < ActiveRecord::Migration[5.2]
+  def up
+    max_id = Place.maximum( :id )
+    batch_size = 10000
+    batch_count = ( max_id / batch_size )
+    for batch_index in ( 0..batch_count ) do
+      start_id = batch_index * batch_size
+      puts start_id
+      Place.where( "id > ? AND id <= ?", start_id, start_id + batch_size )
+        .update_all( "admin_level = admin_level * 10" )
+    end
+
+    max_id = User.maximum( :id )
+    batch_size = 1000
+    batch_count = ( max_id / batch_size )
+    for batch_index in ( 0..batch_count ) do
+      start_id = batch_index * batch_size
+      puts start_id
+      User.where( "id > ? AND id <= ?", start_id, start_id + batch_size )
+        .update_all( "lat_lon_acc_admin_level = lat_lon_acc_admin_level * 10" )
+    end
+  end
+
+  def down
+    max_id = Place.maximum( :id )
+    batch_size = 10000
+    batch_count = ( max_id / batch_size )
+    for batch_index in ( 0..batch_count ) do
+      start_id = batch_index * batch_size
+      puts start_id
+      Place.where( "id > ? AND id <= ?", start_id, start_id + batch_size )
+        .update_all( "admin_level = admin_level / 10" )
+    end
+
+    max_id = User.maximum( :id )
+    batch_size = 1000
+    batch_count = ( max_id / batch_size )
+    for batch_index in ( 0..batch_count ) do
+      start_id = batch_index * batch_size
+      puts start_id
+      User.where( "id > ? AND id <= ?", start_id, start_id + batch_size )
+        .update_all( "lat_lon_acc_admin_level = lat_lon_acc_admin_level / 10" )
+    end
+  end
+end
```

Comment:
This looks pretty chunky. Any estimate on how long it will take?

Reply:
The place one is pretty fast. Users is a bit longer, which is why I'm using a smaller batch size of 1000 so the update queries don't take too long. I don't have an exact time - I'm hoping to get that when running on staging. But I suspect it'll be on the order of a few minutes - less than 15 minutes total

Reply:
Ah, so you're expecting to do this with some downtime?

Reply:
I wasn't planning on doing this with downtime, but we can if you think that'll be preferred. There will be a short period of time when the admin_levels in the DB are updated, but the code the apps are using don't reflect the latest admin_levels since we'll migrate before restarting the apps. That may result in a few small issues for things like the dashboards of users w/ no activity, but I figured that downtime would be more disruptive. Now that I think of it, it's probably best to update the users admin_level first since that'll take a little longer and that change is least likely to result in problems, so I'll change the order in this migration

Reply:
Hm, well I guess if things don't get locked up and Capistrano isn't going to hang during a 15 min deploy then this sounds ok to me.